### ISSUE TITLE
Reminder. Use Symfony\Component\Cache\Adapter\AdapterInterface instead of legacy Cache in CategoryDisplayNameBuilder

### ISSUE THREAD

**Issue by zuk3975** [PrestaShop/PrestaShop#28427]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Reminder raised from 
https://github.com/PrestaShop/PrestaShop/pull/28141#discussion_r863699741

### Describe the solution you'd like

Refactor CategoryDisplayNameBuilder to use Symfony\Component\Cache\Adapter\AdapterInterface instead of legacy

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by HanaRebaiQA** [comment]
Hello @zuk3975 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. 

It's great for your PR.

Thank you!

**Comment by Hlavtox** [comment]
cc @Codencode Isn't this the thing you are implementing now?

**Comment by Codencode** [comment]
Yes, that's the PR: https://github.com/PrestaShop/PrestaShop/pull/39888
However, it's "blocked" because they need to decide how to proceed.
I've linked my PR to this issue so we can keep track of it!
